dnf/db/group.py

```diff
@@ -273,41 +275,63 @@ def add_upgrade(self, new, old, obsoleted=None):
         ti_old = self.new(old, libdnf.transaction.TransactionItemAction_UPGRADED, replaced_by=ti_new)
         self._add_obsoleted(obsoleted, replaced_by=ti_new)
 
+    def _test_fail_safe(self, hdr, pkg):
+        if pkg._from_cmdline:
+            return False
+        if pkg.repo.module_hotfixes:
+            return False
+        try:
+            if hdr['modularitylabel'] and not pkg._is_in_active_module():
+                logger.critical(_("No available modular metadata for modular package '{}', "
+                                  "it cannot be installed on the system").format(pkg))
+                return True
+        except ValueError:
+            return False
+        return False
+
     def _populate_rpm_ts(self, ts):
         """Populate the RPM transaction set."""
+        modular_problems = 0
 
         for tsi in self:
             if tsi.action == libdnf.transaction.TransactionItemAction_DOWNGRADE:
                 hdr = tsi.pkg._header
+                modular_problems += self._test_fail_safe(hdr, tsi.pkg)
```

Comment:
Please change it.
Increasing an integer value by adding (+=) a boolean value is confusing.lease adding boolean option is confusing.